presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -100,10 +101,18 @@ public ConnectorPartitionResult getPartitions(ConnectorTableHandle tableHandle,
                         log.warn("No leader for partition %s/%s found!", metadata.topic(), part.partitionId());
                     }
                     else {
+                        // make sure the first broker is the partition leader
+                        List<HostAddress> isrList = part.isr().stream().map(KafkaSplitManager::brokerToHostAddress).collect(toList());
+                        HostAddress partitionLeader = HostAddress.fromParts(leader.host(), leader.port());
+                        if (!isrList.get(0).getHostText().equals(leader.host())
+                                || isrList.get(0).getPort() != leader.port()) {
+                            isrList.remove(partitionLeader);
```

Comment:
this may or may not work. From the javadoc of `toList()`: There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned. You need to create a copy above if you want to modify the list.